ocaml/typing/typedecl.ml

```diff
@@ -1249,16 +1255,16 @@ module Element_repr = struct
          We may revisit this decision later when we know better whether we want
          flambda2 to unbox for 32 bit platforms.
       *)
-      | Immediate64 -> Value_element
-      | Value -> Value_element
-      | Immediate -> Imm_element
-      | Float64 -> Unboxed_element Float64
-      | Float32 -> Unboxed_element Float32
-      | Word -> Unboxed_element Word
-      | Bits32 -> Unboxed_element Bits32
-      | Bits64 -> Unboxed_element Bits64
-      | Void -> Element_without_runtime_component { loc; ty }
-      | Any ->
+      | Some Value, (Internal | External64) ->
+        Value_element
+      | Some Value, External -> Imm_element
+      | Some Float64, _ -> Unboxed_element Float64
+      | Some Float32, _ -> Unboxed_element Float32
+      | Some Word, _ -> Unboxed_element Word
+      | Some Bits32, _ -> Unboxed_element Bits32
+      | Some Bits64, _ -> Unboxed_element Bits64
+      | Some Void, _ -> Element_without_runtime_component { loc; ty }
+      | None, _ ->
           Misc.fatal_error "Element_repr.classify: unexpected Any"
```

Comment:
```suggestion
          Misc.fatal_error "Element_repr.classify: unexpected abstract layout"
```

Just for future-proofing! (Users won't be able to declare abstract layouts directly. But the layout that's part of an abstract kind will be an abstract layout.)

Reply:
done